### PR TITLE
Simplify trivial abs

### PR DESCRIPTION
**Pull request by zasdfgbnm**
A small incremental step for https://github.com/NVIDIA/Fuser/issues/421

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

File: csrc/expr_simplifier.cpp [hunk 1]
```diff
@@ -1677,16 +1677,22 @@ Val* eliminateTrivialComputation(Val* value, const Context& context) {
...
       if (uop_in != nullptr) {
         auto optype_in = uop_in->getUnaryOpType();
         if (optype == optype_in) {
           return uop_in->in();
         }
       }
+    } else if (optype == UnaryOpType::Abs) {
+      // abs(x) -> x if x >= 0
+      auto abs_in = uop->in();
+      if (prove::isNonNegative(abs_in, context)) {
```
Comment (naoyam) [inline csrc/expr_simplifier.cpp:1693]: Can this also simplify `abs(x)` to `-x` where `x <= 0`?

Comment (zasdfgbnm) [reply csrc/expr_simplifier.cpp:1693]: No, we can not do that yet. Currently, I don't see any use case for this. It is not hard to implement it, just
```C++
else if (auto neg_in = IrBuilder::negExpr(abs_in); prove::isNonNegative(neg_in, context)) {
  return neg_in;
}
```
But the `prove::isNonNegative` is not free, so I prefer not to do it if I don't see any use case. In the future, if we refactor our system to know the sign of `abs_in` in one pass regardless of if it is positive or negative, then we should add that missing simplification.

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**